common/pom.xml

```diff
@@ -50,6 +50,11 @@
             <version>${project.version}</version>
             <scope>compile</scope>
         </dependency>
+        <!--compileOnly 'org.projectlombok:lombok:1.18.12'-->
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
```

Comment:
We need to make a decision on Lombok. Personally I am open to using it, but the team previously decided to phase it out and to use purely AutoValue. Apparently they are not compatible.

Reply:
Right, so I think AutoValue is more heavily used, since it's everywhere in Beam and I also applied it a lot in ingestion. So I changed to AutoValue.

Reply:
Ok, so lets continue to use Lombok for the time being. I hope this doesnt interfere with the rest of the code base.

Reply:
"as key" I meant for grouping operations of course

Reply:
ok, I guess @zhilingc is right. I could use `SchemaCoder`. Of course not through decorator `@DefaultSchema()` since it's common module and we don't want to have dependency on beam here. But that's still feasible. I guess we can move towards AutoValue.